lib/view_component/compiler.rb

```diff
@@ -48,12 +48,17 @@ def compile(raise_errors: false, force: false)
 
         redefinition_lock.synchronize do
           component_class.silence_redefinition_of_method("call")
-          # rubocop:disable Style/EvalWithLocation
-          component_class.class_eval <<-RUBY, template.path, template.lineno
+          source = <<-SOURCE
           def call
             #{compiled_inline_template(template)}
           end
-          RUBY
+          SOURCE
+          # rubocop:disable Style/EvalWithLocation
+          if frozen_string_literal
+            component_class.class_eval("# frozen_string_literal: true\n#{source}", template.path, template.lineno - 1)
+          else
```

Comment:
SimpleCov says this line's not tested, just FYI!

Reply:
Thanks for catching that, added a test in b15f939 which should cover the inline template case.